### PR TITLE
Upload coredumps

### PR DESCRIPTION
**Pull request by dasantiago**
Sometimes an application will generate coredumps. This change will capture
those coredumps and make them available in the assets tab in openqa.

The coredumps will be captured as long as the test invokes the function
problem_detection in the post_fail_hook.

### CODE REVIEW

**Comment by dasantiago** [comment]
Test: http://tragicbox.suse.cz/tests/80/file/snapper_cleanup-problem_detection.tar.xz

File: lib/opensusebasetest.pm [hunk 1]
```diff
@@ -103,8 +107,8 @@ done", "binaries-with-missing-libraries.txt", {timeout => 60, noupload => 1});
         clear_console;
     }
 
-    script_run 'tar cvvJf problem_detection_logs.tar.xz *';
-    upload_logs('problem_detection_logs.tar.xz');
+    script_run 'tar cvvJf problem_detection.tar.xz *';
+    upload_logs('problem_detection.tar.xz');
```
Comment (foursixnine) [inline lib/opensusebasetest.pm]: problem_detection sounds a bit too generic... how about using core_dumps?

Comment (dasantiago) [inline lib/opensusebasetest.pm]: @foursixnine 
My idea, was to have the coredumps in the same place where the related logs are.... 

There's also no guarantee that coredumps are generated, so having an archive "core_dumps.tar.xz" and no core dumps just logs doesn't seem a good option to me.

Maybe some other name, like 'logs_and_cores.tar.xz' ?

Comment (foursixnine) [inline lib/opensusebasetest.pm]: I love the possibility to be able to [get to know those cores](https://www.youtube.com/watch?v=YlFXCC4Hfa0)... But back to being serious, 'logs_and_coredumps.xz' sounds great :)

Comment (okurz) [inline lib/opensusebasetest.pm]: I consider coredumps part of logs so the old name is also fine

File: lib/opensusebasetest.pm [hunk 2]
```diff
@@ -271,14 +275,14 @@ Also see poo#18016 for details.
...
     my $virsh_guest = get_required_var('VIRSH_GUEST');
-    type_line_svirt "sed -i \"\\\"s:IPADDR='[0-9.]*/\\([0-9]*\\)':IPADDR='$virsh_guest/\\1':\\\" /etc/sysconfig/network/ifcfg-\*\"", expect => '#';
+    type_line_svirt("sed -i \"\\\"s:IPADDR='[0-9.]*/\\([0-9]*\\)':IPADDR='$virsh_guest/\\1':\\\" /etc/sysconfig/network/ifcfg-\*\"", expect => '#');
     type_string "# output of current network configuration for debugging\n";
-    type_line_svirt "cat /etc/sysconfig/network/ifcfg-\*", expect => '#';
-    type_line_svirt "systemctl restart network",           expect => '#';
-    type_line_svirt "systemctl is-active network",         expect => 'active';
+    type_line_svirt("cat /etc/sysconfig/network/ifcfg-\*", expect => '#');
+    type_line_svirt("systemctl restart network",           expect => '#');
+    type_line_svirt("systemctl is-active network",         expect => 'active');
```
Comment (okurz) [inline lib/opensusebasetest.pm]: wait, why are you changing all these? Clearly an unrelated change, please remove from the commit and either keep it the old way or create another commit with good name

File: lib/opensusebasetest.pm (hunk 1 above)
Comment (dasantiago) [inline lib/opensusebasetest.pm]: I reverted to the old name: problem_detection_logs.tar.xz

@foursixnine @okurz 
Is it ok with you guys?

**Comment by okurz** [comment]
yep, looks fine. I don't know for sure how it's actually uploading now because there is no upload_logs call but I trust you :-)

File: lib/opensusebasetest.pm (hunk 2 above)
Comment (dasantiago) [inline lib/opensusebasetest.pm]: I was changing that because it's a best practice and IMHO, the file shouldn't have the mixed coding styles all over it. ( 99% of the file is invoking subroutines in parenthesis)

Anyway, i reverted those 4 changed lines. I'll create another commit, with those changes.

**Comment by dasantiago** [comment]
My code, just copies the coredumps to the current folder.

At the end of the subroutine, there's the following code


>     script_run 'tar cvvJf problem_detection_logs.tar.xz *';
> 
>     upload_logs('problem_detection_logs.tar.xz');
> 
>     type_string "popd\n";
